lib/ansible/modules/cloud/podman/podman_container.py

```diff
@@ -0,0 +1,1213 @@
+            module.fail_json(msg="Can't pull image %s:\n%s\v%s" % (image, out, err))
+
+
+class PodmanContainer(object):
+    def __init__(self, module, name, params):
+
+        super(PodmanContainer, self).__init__()
+        self.module = module
+        self.name = name
+        self.info = self.get_info()
+
+    @property
+    def exists(self):
+        return self.info != {}
+
+    @property
+    def different(self):
+        # TODO(sshnaidm): implement difference calculation between input vars
+        # and current container to understand if we need to recreate it
+        return True
+
+    @property
+    def running(self):
+        return False if not self.exists else self.info['State']['Running']
+
+    @property
+    def stopped(self):
+        return False if not self.exists else not self.info['State']['Running']
+
+    def get_info(self):
+        rc, out, err = run_podman_command(
+            module=self.module,
+            args=['container', 'inspect', self.name], ignore_errors=True)
+        if rc != 0:
+            rc, out, err = run_podman_command(
+                module=self.module,
+                args=['container', 'exists', self.name], ignore_errors=True)
+            if rc == 0:
+                self.module.fail_json(msg="Can't inspect existing container %s" % self.name)
+            else:
+                return {}
+        else:
+            return json.loads(out)[0]
+
+    def _action(self, action):
+        command = construct_command_from_params(action, self.module.params, self.module)
+        rc, out, err = run_podman_command(
+            module=self.module,
+            args=['container'] + command,
+            ignore_errors=True)
+        if rc != 0:
+            self.module.fail_json(
+                msg="Can't %s container %s: %s" % (action, self.name, err))
+
+    def run(self):
+        self._action('run')
+
+    def delete(self):
+        self._action('delete')
+
+    def stop(self):
+        self._action('stop')
+
+    def start(self):
+        self._action('start')
+
+    def create(self):
+        self._action('create')
+
+    def recreate(self):
+        self.delete()
+        self.run()
+
+    def restart(self):
+        self.stop()
+        self.run()
+
+
+class PodmanManager(object):
+
+    def __init__(self, module, results):
+
+        super(PodmanManager, self).__init__()
+
+        self.module = module
+        self.results = results
+        self.name = self.module.params.get('name')
+        self.executable = self.module.get_bin_path(module.params.get('executable'), required=True)
+        self.image = self.module.params.get('image')
+        self.state = self.module.params.get('state')
+        self.restart = self.module.params.get('force_restart')
+        self.recreate = self.module.params.get('recreate')
+        self.container = PodmanContainer(self.module, self.name, self.module.params)
+        if self.image:
+            ensure_image_exists(self.module, self.image)
```

Comment:
Since this can interrupt the execution, plz move the check closer to the beginning of this method.

Reply:
Fixed